src/android/java/io/jxcore/node/BtToRequestSocket.java

```diff
@@ -170,77 +170,92 @@ public int GetLocalHostPort() {
     }
 
     public String GetLocalHostAddress() {
-        String ret = "";
-        if(localHostSocket != null && localHostSocket.getInetAddress() != null){
-            ret = localHostSocket.getInetAddress().toString();
+        if(localHostSocket == null){
+            return "";
```

Comment:
It's generally considered best practice to return NULL when you don't have a valid response. Empty string is ambiguous since it could just be that the string is supposed to be empty as opposed to "I can't get the address" which is NULL (or if that isn't supposed to happen, an exception).

Also, can't this method be private? Nobody calls it outside of this function.